tests/THTests.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase #-}
+
+module THTests where
+
+import Data.Bool
+import Test.Hspec
+import Test.QuickCheck
+import Language.Haskell.TH.Syntax
+
+import OpenGames.Preprocessor.THSyntax as THS
+import OpenGames.Preprocessor.Preprocessor as Pre
+import OpenGames.Preprocessor.FreeOpenGames as Free
+import OpenGames.Preprocessor.TH
+import OpenGames.Preprocessor.AbstractSyntax as ABS
+
+import OpenGames.Examples.SimpleAuction
+import OpenGames.Examples.Bimatrix
+import OpenGames.Examples.LemonMarket
+import OpenGames.Examples.Source.LemonMarket
+
+import Numeric.Probability.Distribution
+import OpenGames.Engine.BayesianDiagnostics
+
+
+interpretTop ("hello") (THS.compileBlock <$> auction  )
+interpretTop ("mpennies") (THS.compileBlock <$> thMatchingPennies  )
+interpretTop ("mlemon") (THS.compileBlock <$> thLemon  )
+generateGame "qlemon" qlemonMarket
+
+instance Arbitrary Coin where
+  arbitrary = bool Heads Tails <$> arbitrary
+
+main :: IO ()
+main =
+  hspec $ do
```

Comment:
The contents of this file should probably be in another file called "TestCodeGen" and this main function should call the test functions of those other modules